app/services/host_aspects.rb

```diff
@@ -0,0 +1,93 @@
+module HostAspects
+  class Configuration
+    SUBJECTS = [:configuration]
+
+    SUBJECTS.each do |subject|
+      define_method "register_#{subject}".to_sym do |aspect_name, aspect_model = nil, &block|
+        self.send('register', subject, aspect_name, aspect_model, &block)
+      end
+    end
+
+    SUBJECTS.each do |subject|
+      define_method "#{subject}_aspect".to_sym do
+        @aspects_registry[@aspect_subjects[subject]]
+      end
+    end
+
+    def registered_aspects
+      @aspects_registry.try(:dup) || {}
+    end
+
+    private
+
+    def register(subject, aspect_name, aspect_model = nil, &block)
+      @aspects_registry ||= {}
+      @aspect_subjects ||= {}
+
+      if old_subj = @aspect_subjects[subject]
+        throw "Could not register #{aspect_name}, #{old_subj} is already registered as #{subject} aspect."
+      end
+
+      entry = Entry.new(aspect_name, aspect_model, subject)
+
+      entry.instance_eval(&block) if block_given?
+
+      @aspects_registry[entry.name] = entry
+      @aspect_subjects[subject] = entry.name
+    end
+  end
+
+  class Entry
```

Comment:
You are a second person that comments about it, I will separate them, although I have answered Ohad that these classes are monolithic and there is no use in this separation.